sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -359,20 +359,29 @@ class DeviceBinaryImage {
     return Format;
   }
 
-  /// Gets the iterator range over scalar specialization constants in this this
+  /// Gets the iterator range over scalar specialization constants in this
   /// binary image. For each property pointed to by an iterator within the
-  /// range, the name of the property is the specializaion constant symbolic ID
+  /// range, the name of the property is the specialization constant symbolic ID
   /// and the value is 32-bit unsigned integer ID.
   const PropertyRange &getScalarSpecConstants() const {
     return ScalarSpecConstIDMap;
   }
   /// Gets the iterator range over composite specialization constants in this
-  /// this binary image. For each property pointed to by an iterator within the
-  /// range, the name of the property is the specializaion constant symbolic ID
+  /// binary image. For each property pointed to by an iterator within the
+  /// range, the name of the property is the specialization constant symbolic ID
   /// and the value is a list of tuples of 32-bit unsigned integer values, which
-  /// encode scalar specialization constants, that form a composite one.
+  /// encode scalar specialization constants, that form the composite one.
   /// Each tuple consist of ID of scalar specialization constant, its location
   /// within a composite (offset in bytes from the beginning) and its size.
+  /// For example, for the following structure:
+  /// struct A { int a; float b; };
+  /// struct POD { A a[2]; int b; };
```

Comment:
Is it important to say how the "flattening" works? How are unions handled?

Reply:
> Is it important to say how the "flattening" works? 

Not really. Here runtime fully relies on the compiler to provide correct information about elements of a composite spec constant, i.e. having this byte array and those tuples we need to perform particular API calls

> How are unions handled?

Actually, unions are prohibited at the moment, but technically they will be handled in the same manner: as I can see [in this example](https://godbolt.org/z/zasdKE), union is nothing, but a regular structure at LLVM IR level